kiva/agg/src/kiva_graphics_context.h

```diff
@@ -1292,21 +1294,14 @@ namespace kiva
         ScanlineRendererType scanlineRenderer(this->renderer);
 
         const agg24::glyph_cache *glyph = NULL;
-#if defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
-        int required = MultiByteToWideChar(CP_UTF8, 0, text, -1, 0, 0);
-        std::vector<wchar_t> p_(required + 1);
-        MultiByteToWideChar(CP_UTF8, 0, text, -1, &p_[0], required);
-        wchar_t *p = &p_[0];
-#else
-        std::vector <wchar_t> p_(1024);
-        size_t length = mbstowcs(&p_[0], text, 1024);
-        if (length > 1024)
-          {
-            p_.resize (length + 1);
-            mbstowcs(&p_[0], text, length);
-          }
-        wchar_t *p = &p_[0];
-#endif
+
+        // Explicitly decode UTF8 bytes to 32-bit codepoints to feed into the
+        // font API.
+        std::vector<utf8::uint32_t> codepoints;
+        std::vector<utf8::uint32_t>::iterator p;
+        std::string utf8text(text);
+        utf8::utf8to32(utf8text.begin(), utf8text.end(), std::back_inserter(codepoints));
```

Comment:
I believe you can just construct a `utf8::iterator` and avoid having to copy into `codepoints`. Additionally, if you get the length of `text`, I think you can construct the iterator like so: `utf8::iterator<char*> p(text, text, text+length);`. If that works, it would be nice to avoid the allocations.